reconcile/ocm_clusters.py

```diff
@@ -12,12 +12,16 @@
 QONTRACT_INTEGRATION = 'ocm-clusters'
 
 
-def fetch_current_state(clusters):
+def fetch_desired_state(clusters):
     desired_state = {c['name']: {'spec': c['spec'], 'network': c['network']}
                      for c in clusters}
     # remove unused keys
     for desired_spec in desired_state.values():
         desired_spec['spec'].pop('upgrade', None)
+        # remove empty keys in spec
+        empty_keys = [k for k, v in desired_spec['spec'].items() if v is None]
```

Comment:
you can use a dict comprehension and avoid the pop ahead... `{k: n for k, v in d.items() if v is not None }`

Reply:
absolutely!